websocket_service.go

```diff
@@ -8,9 +8,10 @@ import (
 )
 
 var (
-	baseURL         = "wss://stream.binance.com:9443/ws"
-	baseFutureURL   = "wss://fstream.binance.com/ws"
-	combinedBaseURL = "wss://stream.binance.com:9443/stream?streams="
+	baseURL              = "wss://stream.binance.com:9443/ws"
+	baseFutureURL        = "wss://fstream.binance.com/ws"
+	baseFutureTestnetURL = "wss://stream.binancefuture.com/ws"
```

Comment:
Hi @Krisa , I think it's better to add an optional parameter `cfg`, which is an instance of `newWsConfig`, to [WsFutureUserDataServe](https://github.com/adshao/go-binance/blob/735e7f4648600221ddc128083d0a300129991912/websocket_service.go#L287), since the testnet functions are same to mainnet.

Reply:
So that `WsFutureUserDataServe` will be (optionally) initialized like this `WsFutureUserDataServe(listenKey string, handler WsHandler, errHandler ErrHandler, endpoint string)` ?

Reply:
We need to change [wsConfig](https://github.com/adshao/go-binance/blob/master/websocket.go#L15) to `WsConfig`, so that user can initialize it outside the package, and we can add more config params to `WsConfig` later.
The final method will be like:
`WsFutureUserDataServe(listenKey string, handler WsHandler, errHandler ErrHandler, cfg ...WsConfig)`, any suggestion?

Reply:
@adshao I see and then the user will have to do this: 
```
_, _, err = binance.WsFutureUserDataServe(listenKey, wsHandler, errHandler, &binance.WsConfig{
	Endpoint: "wss://stream.binancefuture.com/ws",
})
```

Whether it's easier... as a third option, the URL could be built also in the listenKey and we would need in WsFutureUserDataServe a simple `if` to check whether the listen key starts with `wss://`...

Reply:
Adding `wss://` to `listenKey` will make it confused.
A better solution is that we can move all these `WsxxxServe` methods to [binance.Client](https://github.com/adshao/go-binance/blob/master/client.go#L172) and [binance.futures.Client](https://github.com/adshao/go-binance/blob/master/futures/client.go#L142), so that the `endpoint` parameter can be configured in `Client` instance, and there is no need for `WsxxxServe` methods to care about `WsConfig`.

Reply:
The difference with `Client` being that REST calls are signed on the fly (the `BaseURL` can be changed after initialization), while for Websockets, the function itself establishes the connection. This means that the user would have to build the URL prior to calling the `WsxxxServe` function, and I don't see what would be the difference with `WsConfig` you mentioned earlier. I can understand the redundancy, you are worried about by having for each URL a specific endpoint, yet from an end user perspective, it requires the less "digging" within the library to figure out what to do.

>Adding `wss://` to `listenKey` will make it confused.
True... But we could call `listenKey` something like `listenKeyOrFullURL` as well as add some comments prior to the function to describe the parameters... I'm not saying it's the best solution, but it was just an idea to avoid a breaking change and make initialization easier... The 4th parameter could also be simply the Endpoint, instead of a Struct containing the endpoint.

Reply:
hi @Krisa  Thanks for the update. Right, to make it simple, we don't need to move ws methods to client at this time. The reason to add an optional parameter `WsConfig` is that there may be other parameters, i.e. timeout, as well as endpoint need to be configured in the future, and an optional parameter make it a backward compatible change and will not break current codes.